src/shmem_comm.h

```diff
@@ -49,7 +49,10 @@ void
 shmem_internal_put_nb(shmem_ctx_t ctx, void *target, const void *source, size_t len, int pe,
                       long *completion)
 {
-    if (len == 0) return;
+    if (len == 0) {
+        *completion = 1;
```

Comment:
Wow.. how did it work before? Guess we never checked this before?

Reply:
OFI doesn't use the completion flags, and I think we just never noticed with the Portals build. At some point, the transport API should be updated to have blocking operations, nonblocking operations with no handle (NBI), and if we really care about the cases where `completion` is used nonblocking operations with a transport-defined request object (NBE). With UCX, it would be much cleaner to pass around the UCX request object rather versus a user-supplied completion flag.